### DISCUSSION TITLE
Importing datasets

### DISCUSSION THREAD

**Discussion by Hafixion** [Kotlin/kotlindl#255] [Q&A] [answered]
Hey! I think this might be a really simple problem but, I'm still figuring out the library. Currently I'm trying out importing datasets from csv files using krangl and then fitting a model to them. I decided to try out a multilabel dataset recently, and I'm a bit confused as to how I'd have multiple labels.

My current implementation ends in a `Array<FloatArray>` which corresponds to rows in the csv file. I also have another `Array<FloatArray>` which corresponds to labels. 

I'm using a OnHeapDataset, and the function to create one is
```kt
/**
         * Takes data from consumers [featuresConsumer] and [labelConsumer]
         * to dataset [OnHeapDataset].
         */
        @JvmStatic
        public fun create(
            featuresConsumer: () -> Array<FloatArray>,
            labelConsumer: () -> FloatArray
        ): OnHeapDataset {
            return try {
                val features = featuresConsumer()
                val labels = labelConsumer()

                check(features.size == labels.size) { "The amount of labels is not equal to the amount of images." }

                OnHeapDataset(features, labels)
            } catch (e: IOException) {
                throw AssertionError(e)
            }
        }
```

My question is then as follows, how would I have multiple labels if all it takes in is a single FloatArray?

P.S. I've probably completely misunderstood something, so be kind.

**Comment by zaleslaw** [comment, marked as answer]
The problem exists and a possible solution is described in issue #254 
So at this moment, no way is to implement multilabel dataset.

But if the fix for the multiple regression is pretty simple, if not the same for the multilabel classification, where each label could be presented as a vector of probabilities.

Could you provide more information about the task which you are going to solve? Is it multilabel regression or is it multilabel classification?

**Reply by Hafixion** [reply]
Hey, thanks for clearing it up. I ended up using an older version of KotlinDL to complete my task. The task was multilabel regression, it was an attempt to create a model reaching high accuracies on the titanic dataset. Ended up getting around 85% test accuracy. I assumed I misunderstood something with the API but I did get it done in the end so there's that.

In hindsight, I should've just treated it as a 0-1 binary problem instead of having 2 labels, but I hope support for multilabel outputs gets added soon, the API has been incredibly comfortable to use so far.
